models/classes/class.LtiService.php

```diff
@@ -19,20 +19,26 @@
  * 
  */
 
-use oat\taoLti\models\classes\LtiMessages\LtiErrorMessage;
+use oat\taoLti\models\classes\AbstractLtiService;
 
 /**
  * Basic service to handle everything LTI
  * 
  * @author Joel Bout, <joel@taotesting.com>
+ * @deprecated use AbstractLtiService
  */
 class taoLti_models_classes_LtiService extends tao_models_classes_Service 
 {
 	const LIS_CONTEXT_ROLE_NAMESPACE = 'urn:lti:role:ims/lis/';
 
 	const LTICONTEXT_SESSION_KEY	= 'LTICONTEXT';
-	
+
+    /** @var AbstractLtiService $ltiService */
+	private $ltiService = null;
+
 	protected function __construct() {
+        $this->ltiService = $this->getServiceLocator()->get(AbstractLtiService::SERVICE_ID);
```

Comment:
no calls to getServiceManager in __construct() as there was no opportunity for injection yet